accepted/single-file/extract.md

```diff
@@ -4,112 +4,115 @@ When running the single-file, the host will need to extract some bundled files r
-* If all files are extracted successfully, host writes an *end-marker*. The end-marker will include name and version information for the app's main assembly in plain text, in order to help users in associating install-locations with app versions.
-* If there are failures, host attempts to clean up all contents of the *install-location*
-* Releases the disk lock. 
+- Complex implementation.
+- Unsuitable for customer scenarios that cannot tolerate persistent disk state after the app terminates.
+- Cleanup is manual.
 
-Subsequent runs reuse the extracted files after confirming their successful extraction. However, if the contents of install-location are corrupted post-extraction, they will need to be explicitly cleaned.
+### Current Tools
 
-#### Install Location
+* **Extraction Time**: Mono extracts native libraries at startup. Costura extracts and explicitly loads the native-libraries at startup.
+* **Lifetime**:  Mono and Costura extract native dependencies to temporary files on every run.
 
-By default, the host extracts dependencies to `BASE_PATH/APP_NAME/ID/PERM/`
+## Proposed Solution
 
-- `BASE_PATH` is  `%HOMEPATH%\.dotnet\Apps ` /  `$HOME/.dotnet/Apps`
-- `APP_NAME` is simple the name of the app (the host executable containing embedded data).
-- `ID` is a hash-code based on the host binary to distinguish app version and architecture specifications.
-- `PERM` is `user` for normal runs, and `admin` for elevated runs with appropriate write-permissions.
+For .Net Core 3.0, we propose to implement the persistent-extraction scheme. Based on customer feedback, we can implement other options discussed above.
 
-A different extraction location may be specified via the runtime configuration settings:
+### Extraction Requirements
 
-```json
-{
-    "runtimeoptions": {
-        "bundledFileExtractionLocation": "<path>"
-    }
-}
-```
+- *Reuse*: Extract on first-run, reuse on subsequent runs.
+- *Fault-tolerance*: Recover from failure after partial extraction on first run.
+- *Concurrency*: Handle concurrent launches on first start.
+- *Upgrade*: Each version of the app extracts to a unique location, supporting side-by-side use of multiple versions.
+- *Cleanup*: Users can identify and delete extracted files when the app is no longer needed.
+- *Access control*: Have an option for elevated runs to extract files to admin-only-writable locations.
 
-#### Cleanup
+### Extraction Location
 
-The cleanup of extracted files in the install-location will be manual in this version. We can consider adding `dotnet CLI` commands for cleanup in future. However, future versions are expected to spill fewer artifacts to disk, making the cleanup commands a lower priority feature.
+The desirable characteristics of the extraction directory are:
 
-#### Pros
+- Should be configurable
+- Should be semi-permeant: Not a directory that is lost frequently (ex: on machine reboot), but considered recyclable storage, since any subsequent run can re-extract.
```

Comment:
```suggestion
- Should be semi-permanent: Not a directory that is lost frequently (ex: on machine reboot), but considered recyclable storage, since any subsequent run can re-extract.
```